internal/controller/changetransferpolicy_controller.go

```diff
@@ -155,6 +162,227 @@ func (r *ChangeTransferPolicyReconciler) Reconcile(ctx context.Context, req ctrl
+	logger := log.FromContext(ctx)
+
+	if pullRequestID := activeTrailers[constants.TrailerPullRequestID]; pullRequestID != "" {
+		h.PullRequest.ID = pullRequestID
+	} else {
+		logger.Info("No " + constants.TrailerPullRequestID + " found in trailers")
+	}
+
+	if pullRequestUrl := activeTrailers[constants.TrailerPullRequestUrl]; pullRequestUrl != "" {
+		h.PullRequest.Url = pullRequestUrl
+	} else {
+		logger.Info("No " + constants.TrailerPullRequestUrl + " found in trailers")
+	}
+
+	if timeStr := activeTrailers[constants.TrailerPullRequestCreationTime]; timeStr != "" {
+		if creationTime, err := time.Parse(time.RFC3339, timeStr); err != nil {
+			logger.Error(err, "failed to parse "+constants.TrailerPullRequestCreationTime, "time", timeStr)
+		} else {
+			h.PullRequest.PRCreationTime = metav1.NewTime(creationTime)
+		}
+	} else {
+		logger.Info("No " + constants.TrailerPullRequestCreationTime + " found in trailers")
+	}
+}
+
+// populateCommitStatuses populates the commit statuses for a history entry
+func (r *ChangeTransferPolicyReconciler) populateCommitStatuses(ctx context.Context, h *promoterv1alpha1.History, activeTrailers map[string]string) {
+	activeKeys, proposedKeys := getCommitStatusKeysFromTrailers(ctx, activeTrailers)
+
+	h.Active.CommitStatuses = make([]promoterv1alpha1.ChangeRequestPolicyCommitStatusPhase, 0, len(activeKeys))
+	for _, key := range activeKeys {
+		h.Active.CommitStatuses = append(h.Active.CommitStatuses, promoterv1alpha1.ChangeRequestPolicyCommitStatusPhase{
+			Key:   key,
+			Phase: activeTrailers[constants.TrailerCommitStatusActivePrefix+key+"-phase"],
+			Url:   activeTrailers[constants.TrailerCommitStatusActivePrefix+key+"-url"],
+		})
+	}
+
+	h.Proposed.CommitStatuses = make([]promoterv1alpha1.ChangeRequestPolicyCommitStatusPhase, 0, len(proposedKeys))
+	for _, key := range proposedKeys {
+		h.Proposed.CommitStatuses = append(h.Proposed.CommitStatuses, promoterv1alpha1.ChangeRequestPolicyCommitStatusPhase{
+			Key:   key,
+			Phase: activeTrailers[constants.TrailerCommitStatusProposedPrefix+key+"-phase"],
+			Url:   activeTrailers[constants.TrailerCommitStatusProposedPrefix+key+"-url"],
+		})
+	}
+}
+
+// getCommitStatusKeysFromTrailers extracts the commit status keys from the trailers in the given context.
+func getCommitStatusKeysFromTrailers(ctx context.Context, trailers map[string]string) (activeKeys []string, proposedKeys []string) {
+	logger := log.FromContext(ctx)
+
+	extractKeys := func(prefix string) []string {
+		keys := []string{}
+		for key, trailer := range trailers {
+			if !strings.HasPrefix(key, prefix) {
+				continue
+			}
+			key = strings.TrimPrefix(key, prefix)
+			if key == "" {
+				logger.Info("Skipping empty trailer key", "trailer", trailer)
+				continue
+			}
+			parts := strings.Split(key, "-")
+			if len(parts) < 2 {
+				logger.Info("Skipping trailer with unexpected format", "trailer", trailer)
+				continue
+			}
```

Comment:
Maybe just strings.Cut?

Reply:
Don't think I can the logic is different, I am. just trying to get the key from the trailer key which means since I only have one -url or -phase I strip the last split on - off and then join back to a string could maybe do two specific cuts with know prefixes such as -url or -phase I think whats current is cleaner.